src/build/packages.rs

```diff
@@ -190,127 +191,158 @@ fn read_bsconfig(package_dir: &str) -> bsconfig::T {
+fn read_dependencies<'a>(
+    registered_dependencies_set: &'a mut AHashSet<String>,
+    parent_bsconfig: bsconfig::T,
+) -> Vec<Dependency> {
+    return parent_bsconfig
         .bs_dependencies
         .to_owned()
         .unwrap_or(vec![])
         .iter()
         .filter_map(|package_name| {
-            let package_dir = match PathBuf::from(get_package_dir(package_name, false)).canonicalize() {
-                Ok(dir) => dir.to_string_lossy().to_string(),
-                Err(e) => {
-                    print!(
-                        "{} {} Error building package tree (are node_modules up-to-date?)... \n More details: {}",
-                        style("[1/2]").bold().dim(),
-                        CROSS,
-                        e.to_string()
-                    );
-                    std::process::exit(2)
-                }
-            };
-
-            if !map.contains_key(package_name) {
-                Some(package_dir)
-            } else {
+            if registered_dependencies_set.contains(package_name) {
                 None
+            } else {
+                registered_dependencies_set.insert(package_name.to_owned());
+                Some(package_name.to_owned())
             }
         })
         .collect::<Vec<String>>()
-        // read all bsconfig files simultanously instead of blocking
+        // Read all bsconfig files in parallel instead of blocking
         .par_iter()
-        .map(|package_dir| (package_dir.to_owned(), read_bsconfig(package_dir)))
-        .collect::<Vec<(String, bsconfig::T)>>()
-        .iter()
-        .fold(map, |map, (package_dir, child_bsconfig)| {
-            build_package(
-                map,
-                child_bsconfig.to_owned(),
-                &package_dir,
-                &project_root,
-                bsconfig
-                    .pinned_dependencies
-                    .as_ref()
-                    .map(|p| p.contains(&child_bsconfig.name))
-                    .unwrap_or(false),
-                false,
-            )
-        })
+        .map(|package_name| {
+            let path =
+                match PathBuf::from(helpers::get_relative_package_path(package_name, false)).canonicalize() {
+                    Ok(dir) => dir.to_string_lossy().to_string(),
+                    Err(e) => {
+                        print!(
+                          "{} {} Error building package tree (are node_modules up-to-date?)... \n More details: {}",
+                          style("[1/2]").bold().dim(),
+                          CROSS,
+                          e.to_string()
+                        );
+                        std::process::exit(2)
+                    }
+                };
+
+            let bsconfig = read_bsconfig(&path);
+            let is_pinned = parent_bsconfig
+            .pinned_dependencies
+            .as_ref()
+            .map(|p| p.contains(&bsconfig.name))
+            .unwrap_or(false);
```

Comment:
Is this logic correct? Shouldn't we iterate over `pinned_dependencies` instead?  Also, I'm confused about `dev` dependencies. I don't see us reading them.

Reply:
actually we don't care if a package is pinned, all packages in rewatch are "pinned" so we don't necessarily need to read the `pinned_dependencies` list

Reply:
oh wait we have some special behavior (how we handle warning etc) :)